cupsfilters/catalog.c

```diff
@@ -95,15 +173,33 @@ cfCatalogSearchDir(const char *dirname)
   if (dirname == NULL)
     return (NULL);
 
-  // Check first whether we have an English file and prefer this
-  snprintf(catalogpath, sizeof(catalogpath), "%s/en/cups_en.po", dirname);
-  if (access(catalogpath, R_OK) == 0)
+  if (preferredlocale)
   {
-    // Found
-    catalog = strdup(catalogpath);
-    return (catalog);
+    // Check first for an exact match
+    catalog = cfCatalogSearchDirLocale(dirname, preferredlocale);
+
+    // Check for language match, with any region
+    // Cover both cases, whether locale has region suffix or 
+    strncpy(lang, preferredlocale, strcspn(preferredlocale, "_"));
```

Comment:
Do we have the terminating zero here? Or is lang pre-filled with zeros?

Reply:
I didn't know `strncpy()` didn't append a `NULL` terminator like `snprintf()` does. Fixed it now. Thanks!